lib/js_of_ocaml/dom_html.ml

```diff
@@ -2165,6 +2165,37 @@ class type history =
 
 class type undoManager = object end
 
+class type optionPromise =
+  object
+    method url : js_string t option prop
```

Comment:
You don't want `option` here. `optdef` maybe ?

Reply:
I had a lot of trouble to create the constructor "optionPromise", I "deduce" the 'option' type from the only successful experimentation  tu I indeed hesitate with "optdef" and "opt".

Reply:
an OCaml `option` is represented either by `0` (for `None`) or `[0; x]` (for `Some x`)

Reply:
I don't know how to create a constructor of "optionPromise" (that I'm renaming "shareOption" if the parameters are optdef, it will return:

<<
2416 |   option##.text := text;
         ^^^^^^^^^^^^^
Error: This expression has type
         js_string t optdef prop =
           < get : js_string t optdef; set : js_string t optdef -> unit >
           gen_prop
       but an expression was expected of type
         < set : 'a option -> unit; .. > gen_prop
       Types for method set are incompatible
>>

I haven't found any other way to create an instance of "shareOption" other than a typed "Js.Unsafe.obj" and I don't understood who to use the optdef values here...

Reply:
try `option##.text := Optdef.option text;`